metrics/lib/common.bash

```diff
@@ -95,18 +95,47 @@ check_images()
 # This function performs a docker build on the image names
 # passed in, to ensure that we have the latest changes from
 # the dockerfiles
+build_dockerfile_image()
+{
+	local image="$1"
+	local dockerfile_path="$2"
+
+	echo "docker building $image"
+	if ! docker build --label "$image" --tag "${image}" -< "$dockerfile_path"; then
+		die "Failed to docker build image $image"
+	fi
+}
+
+# This function verifies that the dockerfile version is
+# equal to the test version in order to build the image or
+# just run the test
 check_dockerfiles_images()
 {
 	local image="$1"
 	local dockerfile_path="$2"
+
 	if [ -z "$image" ] || [ -z "$dockerfile_path" ]; then
 		die "Missing image or dockerfile path variable"
-		exit 1;
 	fi
-	echo "docker building $image"
-	if ! docker build --label "$image" --tag "${image}:latest" -< "$dockerfile_path"; then
-		die "Failed to docker build image $image"
-		exit 1;
+
+	# Verify that dockerfile version is equal to test version
+	check_image=$(docker images "$image" -q)
+	if [ -n "$check_image" ]; then
+		# Check image label
+		check_image_version=$(docker image inspect $image | grep -w DOCKERFILE_VERSION | head -1 | cut -d '"' -f4)
+		if [ -n "$check_image_version" ]; then
+			echo "$image is not updated"
+			build_dockerfile_image "$image" "$dockerfile_path"
+		else
+			# Check dockerfile label
+			dockerfile_version=$(grep DOCKERFILE_VERSION $dockerfile_path | cut -d '"' -f2)
+			if [ "$dockerfile_version" -ne "$check_image_version" ]; then
```

Comment:
I think this is still comparing two values containing periods? As mentioned in the last review, `-ne` won't handle that:

```
a=2
b=3.0
[ "$a" -ne "$b" ] && echo not equal
[: integer expression expected: 3.0
```

But you can use a string compare if you only need to know if they are not the same:

```
a=2
b=3.0
[ "$a" != "$b" ] && echo not equal
not equal
```